### ISSUE TITLE
essentials-project does not support spring-dev-tools hot reload functionality

### ISSUE THREAD

**Issue by artyomsv** [cloudcreate-dk/essentials-project#100]
Hi, I`m playing with the library and found one issue. In case spring-dev-tools reload functionality is enabled development classes are being loaded with **RestartClassLoader** instead of default one **AppClassLoader** . This leads to a scenario when **AnnotatedCommandHandler.getHandlerMethod** line with  **isAssignableFrom**  check cannot correctly verify the same command class and an exception is being thrown:
```
dk.cloudcreate.essentials.reactive.command.NoCommandHandlerFoundException: Couldn't find a CommandHandler that can handle a command of type 'com.example.demo.commands.StartCommand'
```
![spring-dev-tools](https://github.com/user-attachments/assets/42f8db67-a694-47cc-a5a7-f70f1e9f058f)

The temporary workaround is to disable **restart** functionality, but I think it is preferable to find a proper solution
```java
@SpringBootApplication
public class DemoApplication {

	public static void main(String[] args) {
		System.setProperty("spring.devtools.restart.enabled", "false");
		SpringApplication.run(DemoApplication.class, args);
	}

}
```

Spring Boot application with this scenario can be found here: https://github.com/artyomsv/event-modeling-questions
That project has **[rest-client.rest](https://github.com/artyomsv/event-modeling-questions/blob/master/rest-client.rest)** rest client script file that can be used to make a rest calls to the app.

**Comment by cloudcreate-dk** [comment]
Thanks for reporting the issue. This problem should be solved in version 0.40.18.

**Comment by cloudcreate-dk** [comment]
Quick feedback on the referenced example: 
The `ApiController` sends the command using `commandBus.sendAndDontWait(new StartCommand(aggregateId));` which results in the command being queued in the `CommandBus` and afterwards asynchronously delivered to the `CommandProcessor` in a new `UnitOfWork`/Transaction.
The duplicate check in `CommandProcessor`'s `public void handle(StartCommand cmd)` throws an exception in case the command has already been started, which will result in the command being retried multiple times and finally marked as a dead-letter-message.
For asynchronous processing it's better to silently ignore duplicate commands (which is something that can always happen when commands/messages are queued).
